gdpr/models.py

```diff
@@ -121,32 +149,63 @@ class LegalReason(SmartModel):
         verbose_name=_('purpose'),
         null=False,
         blank=False,
-        choices=(
-            (purpose_slug, purpose_class.name)
-            for purpose_slug, purpose_class in purposes_map.items()
-        ),
-        max_length=100
+        max_length=100,
+        db_index=True
     )
     source_object_content_type = models.ForeignKey(
         ContentType,
         verbose_name=_('source object content type'),
         null=False,
-        blank=False
+        blank=False,
+        on_delete=models.DO_NOTHING
     )
     source_object_id = models.TextField(
         verbose_name=_('source object ID'),
-        null=False, blank=False
+        null=False, blank=False,
+        db_index=True
     )
     source_object = GenericForeignKey(
         'source_object_content_type', 'source_object_id'
     )
 
     @property
-    def purpose(self):
-        return purposes_map.get(self.purpose_slug, None)
+    def purpose(self) -> Type["AbstractPurpose"]:
+        return purpose_register.get(self.purpose_slug, None)
+
+    def _anonymize_obj(self, *args, **kwargs):
+        purpose_register[self.purpose_slug]().anonymize_obj(self.source_object, self, *args, **kwargs)
+
+    def _deanonymize_obj(self, *args, **kwargs):
+        purpose_register[self.purpose_slug]().deanonymize_obj(self.source_object, *args, **kwargs)
+
+    def _expirement(self):
```

Comment:
Proč je to venku z `expire`?

Reply:
Původní myšlenka byla, že metoda `expire` se nebude volat z periodického checku vypršených LegalReason.